compile/compile.go

```diff
@@ -210,17 +219,72 @@ func (c *Compiler) WithMetadata(metadata *map[string]interface{}) *Compiler {
 	return c
 }
 
+func addEntrypointsFromAnnotations(c *Compiler, ar []*ast.AnnotationsRef) error {
+	for _, ref := range ar {
+		scope := ref.Annotations.Scope
+		entrypoint := ast.EmptyRef()
```

Comment:
Why do we need this? For some reason `var entrypoint ast.Ref` seems to be insufficient..? 🤔